### ISSUE TITLE
Large shadows and borders truncated

### ISSUE THREAD

**Issue by TheOneric** [libass/libass#391]
Shadows with a large offset are truncated in libass, but not MPC-HC. THis happens for BordStyle 1 and 3.
In case of borders overlapping with another line this also seems to be the case.

Sample 1 (Shadows; was rendered with a 640x480 video):
MPC-HC:
![grafik](https://user-images.githubusercontent.com/1668471/80916479-3cce9a00-8d59-11ea-88ef-5a84ec4db449.png)

Libass:
![grafik](https://user-images.githubusercontent.com/1668471/80916458-1ad51780-8d59-11ea-9130-f28e4231467c.png)

```
﻿[Script Info]
; Script generated by Aegisub 9111-dev-5dd58aca8
; http://www.aegisub.org/
Title: Default Aegisub file
ScriptType: v4.00+
WrapStyle: 0
PlayResX: 1920
PlayResY: 1080
ScaledBorderAndShadow: yes
YCbCr Matrix: None

[Aegisub Project Garbage]
Last Style Storage: Default

[V4+ Styles]
Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, Alignment, MarginL, MarginR, MarginV, Encoding
Style: DefaultB,DejaVu Sans,60,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,1,30,0,2,30,30,30,1
Style: DefaultS,DejaVu Sans,60,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,1,0,50,2,30,30,30,1
Style: DefaultBS,DejaVu Sans,60,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,1,30,50,2,30,30,30,1

[Events]
Format: Layer, Start, End, Style, Name, MarginL, MarginR, MarginV, Effect, Text
Dialogue: 0,0:00:00.00,0:00:05.00,DefaultB,,0,0,0,,Lorem Ipsum is simply dummy text of the printing and typesetting industry.
Dialogue: 0,0:00:02.00,0:00:07.00,DefaultB,,0,0,0,,Neque porro quisquam est qui dolorem ipsum quia dolor.
Dialogue: 0,0:00:04.00,0:00:09.00,DefaultB,,0,0,0,,Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium.
Dialogue: 0,0:00:10.00,0:00:15.00,DefaultS,,0,0,0,,Lorem Ipsum is simply dummy text of the printing and typesetting industry.
Dialogue: 0,0:00:12.00,0:00:17.00,DefaultS,,0,0,0,,Neque porro quisquam est qui dolorem ipsum quia dolor.
Dialogue: 0,0:00:14.00,0:00:19.00,DefaultS,,0,0,0,,Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium.
Dialogue: 0,0:00:20.00,0:00:25.00,DefaultBS,,0,0,0,,Lorem Ipsum is simply dummy text of the printing and typesetting industry.
Dialogue: 0,0:00:22.00,0:00:27.00,DefaultBS,,0,0,0,,Neque porro quisquam est qui dolorem ipsum quia dolor.
Dialogue: 0,0:00:24.00,0:00:29.00,DefaultBS,,0,0,0,,Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium.
```

Sample 2 (Top lines: border)
MPC-HC:
![grafik](https://user-images.githubusercontent.com/1668471/80916557-b6668800-8d59-11ea-8c74-8c29946feb5f.png)

Libass:
![grafik](https://user-images.githubusercontent.com/1668471/80916561-bbc3d280-8d59-11ea-88f9-44c5e9053076.png)

*(Note that this sample also shows differences related to #41 and #304)*
```
﻿[Script Info]
; Script generated by Aegisub 9111-dev-5dd58aca8
; http://www.aegisub.org/
Title: Default Aegisub file
ScriptType: v4.00+
WrapStyle: 0
ScaledBorderAndShadow: yes
YCbCr Matrix: None
PlayResX: 1280
PlayResY: 720

[Aegisub Project Garbage]
Video File: ?dummy:23.976000:400:1280:720:9:169:225:
Video AR Value: 1.777778
Video Zoom Percent: 1.000000

[V4+ Styles]
Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, Alignment, MarginL, MarginR, MarginV, Encoding
Style: NoSNoB,DejaVu Sans,48,&H00000000,&H000000FF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,1,0,0,2,10,10,10,1

[Events]
Format: Layer, Start, End, Style, Name, MarginL, MarginR, MarginV, Effect, Text
Dialogue: 0,0:00:00.00,0:00:05.00,NoSNoB,,0,0,0,,{\an1}nnnw■wnnn\N\Nfffw■wggg\Nfffw■wfff
Dialogue: 0,0:00:00.00,0:00:05.00,NoSNoB,,0,0,0,,{\an3}fffw■wggg\N\Nfffw■wfff\Nnnnw■wnnn
Dialogue: 0,0:00:00.00,0:00:05.00,NoSNoB,,0,0,0,,{\an6}\N■■■■■■■■■■■■■■■
Dialogue: 0,0:00:00.00,0:00:05.00,NoSNoB,,0,0,0,,{\an4} \N■■■■■■■■■■■■■■■
Dialogue: 0,0:00:00.00,0:00:05.00,NoSNoB,,0,0,0,,{\an7}■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■
Dialogue: 0,0:00:00.00,0:00:05.00,NoSNoB,,0,0,0,,{\an7\bord64}■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■\n■
```

**Comment by astiob** [comment]
Seems related to #121: it seems we clip non-explicitly-positioned lines against the frame _first,_ as if they’re the only line in the frame and positioned at the very bottom/top of the screen without any colliding lines, and only then perform collision detection and move lines up and down to avoid overlaps.

**Comment by TheOneric** [comment]
Merged into #121 .
